ioctl/cmd/ins/insregister.go

```diff
@@ -0,0 +1,352 @@
+	BytesArray, _ := abi.NewType("bytes[]", "", nil)
+	Bool, _ := abi.NewType("bool", "", nil)
+	Uint16, _ := abi.NewType("uint16", "", nil)
+	args := abi.Arguments{
+		{Type: Bytes32, Name: "name"},
+		{Type: Address, Name: "owner"},
+		{Type: Uint256, Name: "duration"},
+		{Type: Bytes32, Name: "secret"},
+		{Type: Address, Name: "resolver"},
+		{Type: BytesArray, Name: "data"},
+		{Type: Bool, Name: "reverseRecord"},
+		{Type: Uint16, Name: "ownerControlledFuses"},
+	}
+
+	packed, err := args.Pack(lable, owner, duration, secret, resolver, data, reverseRecord, ownerControlledFuses)
+	if err != nil {
+		return [32]byte{}, err
+	}
+
+	hash := sha3.NewLegacyKeccak256()
+	hash.Write(packed)
+	result := [32]byte{}
+	copy(result[:], hash.Sum(nil))
+
+	return result, nil
+}
+
+func commit(name string, data []byte, ioController address.Address, controllerAbi abi.ABI) error {
+	commitment, err := makeCommitment(
+		name,
+		common.HexToAddress(owner),
+		duration,
+		secret,
+		common.HexToAddress(resolver),
+		[][]byte{data},
+		true,
+		0,
+	)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("commit commitment for %s: %s ...\n", name, hex.EncodeToString(commitment[:]))
+
+	commitData, err := controllerAbi.Pack("commit", commitment)
+	if err != nil {
+		return output.NewError(output.ConvertError, "failed to pack commit arguments", err)
+	}
+	err = action.Execute(ioController.String(), big.NewInt(0), commitData)
+	if err != nil {
+		return output.NewError(output.UpdateError, "failed to commit commitment", err)
+	}
+	return nil
+}
+
+func register(args []string) error {
+	name := args[0]
+
+	if owner == "" {
+		signer, err := action.Signer()
+		if err != nil {
+			return output.NewError(output.AddressError, "failed to get signer address", err)
+		}
+		signerAddr, err := address.FromString(signer)
+		if err != nil {
+			return output.NewError(output.AddressError, "failed to convert signer address", err)
+		}
+		owner = signerAddr.Hex()
+	}
+
+	hash, err := nameHash(name)
+	if err != nil {
+		return output.NewError(output.SerializationError, "failed to compute namehash", err)
+	}
+
+	setAddrAbi, err := abi.JSON(strings.NewReader(setAddrABI))
+	if err != nil {
+		return output.NewError(output.SerializationError, "failed to unmarshal abi", err)
+	}
+	setAddrData, err := setAddrAbi.Pack("setAddr", hash, common.HexToAddress(owner))
+	if err != nil {
+		return output.NewError(output.ConvertError, "failed to pack setAddr arguments", err)
+	}
+
+	ioController, err := address.FromHex(controller)
+	if err != nil {
+		return output.NewError(output.AddressError, "failed to convert controller address", err)
+	}
+	controllerAbi, err := abi.JSON(strings.NewReader(controllerABI))
+	if err != nil {
+		return output.NewError(output.SerializationError, "failed to unmarshal abi", err)
+	}
+
+	if err = commit(name, setAddrData, ioController, controllerAbi); err != nil {
+		return err
+	}
+
+	fmt.Printf("sleep for activation commitment ...\n")
+	time.Sleep(60 * time.Second)
```

Comment:
does it need 60 seconds long?
```
fmt.Printf("wait 60 seconds for activation to be committed ...\n")
```

Reply:
The contract limits the submission of the commitment to 60 seconds before the name can be registered

Reply:
fired by the events from the chain would be better